app/models/concerns/has_upload_references.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+module HasUploadReferences
```

Comment:
Very minor, but maybe this would be better as `HasPostUploadReferences` ? Since it's only used on post-related models, and it might muddy the waters a bit making people think they need to use this concern when dealing with upload references generally, where in reality they just have to make a record in the table.